frontend/src/components/ColorMapLegend.tsx

```diff
@@ -6,19 +6,24 @@ export default function ColorMapLegend({map}: {map: ColorMap}) {
   const min = map[0][0]
   const max = map[map.length - 1][0]
   const normalizeValue = (v) => (v - min) / (max - min)
-
+  if( typeof ColorMapLegend.counter == 'undefined' ) {
+     ColorMapLegend.counter = 0;
+  }
+  ColorMapLegend.counter++;
+  ColorMapLegend.id="gradient"+ColorMapLegend.counter;
+  ColorMapLegend.url="url(#"+ColorMapLegend.id+")";
```

Comment:
This is not very ideomatic react. It will cause all sorts of issues. I guess that changing the ID or contents of a gradient doesn't matter to the browser, so you should use `useMemo` instead to generate a unique ID once for the component instance.

```javascript
const gradientId = useMemo(() => `gradient${Math.floor(Math.random() * 1000000)}`, []);
const gradientUrl = `url(#{graidentId})`;
```

Reply:
Thank you for introducing me to´´useMemo`` as well as the fact that attaching attributes to the function can be problematic for react.